lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -61,6 +61,10 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
         canonicalID, cipherBundle, request, isDeleteMarker, streamingV4Params,
         log, callback) {
     const size = isDeleteMarker ? 0 : request.parsedContentLength;
+    // although the request method may actually be 'DELETE' if creating a
+    // delete marker, for our purposes we consider this to be a 'PUT'
+    // operation
+    const requestMethod = 'PUT';
```

Comment:
I feel like doing this might mess with this line in data.batchDelete: 
https://github.com/scality/S3/blob/master/lib/data/wrapper.js#L229
Or is the idea that, if using AWS backend, we won't have to go through that batchDelete to delete objects anymore?

Reply:
I think we already opted to not go through with the batchDelete if using AWS backend for cleaning up old data for normal PUTs. The idea is to do the same when creating delete markers, even though the request.method in that case is "DELETE" instead of "PUT", to be consistent.